### PR TITLE
adding dark mode

### PR DESCRIPTION
**Pull request by gersongams**
Hi Cristian, I added the dark mode without making any big change (just separate the styles in a different file), I hope it works, DM if you need something else.

### CODE REVIEW

File: copy-assets.js [hunk 1]
```diff
@@ -3,11 +3,30 @@ const path = require('path');
 const { exec } = require('child_process');
 
 const options = [
-    { from: path.resolve(__dirname, './src/index.html'), to: path.resolve(__dirname, './public/') },
-    { from: path.resolve(__dirname, './src/manifest.json'), to: path.resolve(__dirname, './public/') },
-    { from: path.resolve(__dirname, './src/sw.js'), to: path.resolve(__dirname, './public/') },
-    { from: path.resolve(__dirname, './assets'), to: path.resolve(__dirname, './public/') },
-    { from: path.resolve(__dirname, './favicon.ico'), to: path.resolve(__dirname, './public/') }
+  {
```
Comment (cristianbote) [inline copy-assets.js]: I think this is not intended, right?

File: src/index.html [hunk 2]
```diff
@@ -12,6 +12,7 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0">
     <meta name="theme-color" content="#ffb400">
     <title>Vanilla Hacker News PWA</title>
+    <link rel="stylesheet" href="style.css">
```
Comment (cristianbote) [inline src/index.html]: This is a great idea and usually it makes it all together better to work with! In this case though, my goal is to minimize as much as possible external loading resources, so the browser would load and parse everything together.

File: src/index.html [hunk 3]
```diff
@@ -585,6 +47,9 @@
                     <span class="ver">{{version}}</span>
                 </a>
             </nav>
+            <button class="dark-mode-trigger" onclick="localStorage.setItem('mode', (localStorage.getItem('mode') || 'dark') === 'dark' ? 'light' : 'dark'); localStorage.getItem('mode') === 'dark' ? document.querySelector('body').classList.add('dark') : document.querySelector('body').classList.remove('dark')">
```
Comment (cristianbote) [inline src/index.html]: 👍 love the idea! 😃

File: src/index.html [hunk 4]
```diff
@@ -585,6 +47,9 @@
...
+                <img width="16px" height="16px" src='data:image/svg+xml;utf8,<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="adjust" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" class="svg-inline--fa fa-adjust fa-w-16 fa-2x"><path fill="currentColor" d="M8 256c0 136.966 111.033 248 248 248s248-111.034 248-248S392.966 8 256 8 8 119.033 8 256zm248 184V72c101.705 0 184 82.311 184 184 0 101.705-82.311 184-184 184z" class=""></path></svg>' alt="dark mode" />
```
Comment (cristianbote) [inline src/index.html]: Should we use the svg directly here? Is there something else going on that won't let use use the svg?

File: src/style.css [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+.loading .donut {
+  display: block;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  z-index: 1;
+}
+
+.dark {
```
Comment (cristianbote) [inline src/style.css]: I was thinking we should use `@media (prefers-color-scheme: dark)` here so we could have dark mode for users who already have opted in for it, system wide.

**Review by cristianbote — CHANGES_REQUESTED** [review]
Hey @gersongams! Awesome stuff! I'm added a few notes. Let me know what you think! This feels soooo awesome! 🎉 💯

**Comment by cristianbote** [comment]
This one handles #18

File: src/index.html (hunk 4 above)
Comment (gersongams) [reply src/index.html]: I tried to do it as it was above with the github logo 😅😂.

File: src/index.html (hunk 2 above)
Comment (gersongams) [reply src/index.html]: If you want I can do it in the same html.

Comment (cristianbote) [reply src/index.html]: That would be awesome! Thank you!

File: src/index.html (hunk 4 above)
Comment (cristianbote) [reply src/index.html]: Oopsie! 😄 I guess it's fine as it is then -- sorry forgot the previous stuff.

**Comment by gersongams** [comment]
Hi @cristianbote I added the @media (prefers-color-scheme: dark) and remove the button, and also set the styles in the same html file.

**Review by cristianbote — APPROVED** [review]
You rock! Such an elegant fix! 🎉 Awesome job! 😄 Thank you sooo much for helping out!

**Pull request merged**